lib/created_message_queue_handler.ts

```diff
@@ -247,15 +261,36 @@ export async function handleMessage(
   // query succeeded, we may have a profile
   const maybeProfile = errorOrMaybeProfile.value;
 
-  // whether the recipient wants us to store the message content
-  const isMessageStorageEnabled = maybeProfile.exists(
-    profile => profile.isInboxEnabled === true
+  // channels ther user has blocked for this sender service
+  const blockedChannels = maybeProfile
+    .chain(profile =>
+      fromNullable(profile.blockedChannels).map(
+        (bc: IProfileBlockedChannels) =>
+          bc[newMessageWithContent.senderServiceId]
+      )
+    )
+    .getOrElse(new Set());
+
+  winston.debug(
+    "handleMessage|Blocked Channels(%s): %s",
+    newMessageWithContent.fiscalCode,
+    JSON.stringify(blockedChannels)
   );
 
   //
   //  Inbox storage
   //
 
+  // check if the user has blocked inbox message storage from this service
+  const userHasBlockedMessageStorage = blockedChannels.has(
+    BlockedChannelEnum.INBOX
```

Comment:
how is this option exposed to the user?

Reply:
we need to clarify this from a UX point of view: does it makes sense to disable WEBHOOK and not INBOX? shouldn't they go together?

Reply:
another option is to replce "INBOX" with "ALL" and block all notification channels *and* the inbox.
in other words, if the inbox (message storage) is blocked, then every channel is blocked. 

cc @matteodesanti

Reply:
yes, conceptually the inbox is not a notification channel (the inbox is orthogonal to the notifications) - what do you think about something like:

```ts
type BlockedInboxOrChannels = { [k: Channel]: true } | "INBOX";
```
`INBOX` implies inbox and all channels are blocked

Reply:
nice but note that openapi 2.0 spec (the version we use) does not support "oneOf" in the schema and so doesn't our typescript generator (italia-utils). it means that

1. we cannot have that type generated from the swagger specs
2. actually there's no way to enforce this in clients

Reply:
but that's an opportunity to add a new feature to our tool! 😸

Reply:
as a workaround we can just settle to union("inbox", "webhook", "email") and handle the logic server side (more or less like the actual logic but implying "webhook", "email" in case inbox is set)

Reply:
yes, less elegant (the type doesn't fully describe the logic) but I guess that's the only choice